api/src/main/java/net/wesjd/anvilgui/AnvilGUI.java

```diff
@@ -212,6 +217,44 @@ private void closeInventory(boolean sendClosePacket) {
         }
     }
 
+    /**
+     * Updates the title of the AnvilGUI to the new one.
+     *
+     * @param literalTitle The title to use as literal text
+     * @param preserveRenameText Whether to preserve the entered rename text
+     * @throws IllegalArgumentException when literalTitle is null
+     * @see Builder#title(String)
+     */
+    public void setTitle(String literalTitle, boolean preserveRenameText) {
+        Validate.notNull(literalTitle, "literalTitle cannot be null");
+        setTitle(WRAPPER.literalChatComponent(literalTitle), preserveRenameText);
+    }
+
+    /**
+     * Updates the title of the AnvilGUI to the new one.
+     *
+     * @param json The json used to parse into a rich chat component
+     * @param preserveRenameText Whether to preserve the entered rename text
+     * @throws IllegalArgumentException when json is null
+     * @see Builder#jsonTitle(String)
+     */
+    public void setJsonTitle(String json, boolean preserveRenameText) {
+        Validate.notNull(json, "json cannot be null");
+        setTitle(WRAPPER.jsonChatComponent(json), preserveRenameText);
+    }
+
+    private void setTitle(Object title, boolean preserveRenameText) {
+        if (!WRAPPER.isCustomTitleSupported()) {
+            return;
```

Comment:
Should we indicate to the library user that the title set failed? I understand that we should not throw an exception as many people would like to use the library cross platform, but I think returning a boolean of success status could suffice.

Reply:
I think it would make more sense if we would expose the state of custom title support. Because at the moment the developer will have no direct indicator that the builder methods for custom titles were successful either.

Reply:
That is a fair point. For now, I suppose we will just have to stick with what we have. Unsure whats the best way to go about that.